### DISCUSSION TITLE
Disable all rules, then opt-in to the ones I want

### DISCUSSION THREAD

**Discussion by spaceunifyfifty** [scalameta/scalafmt#4166] [Q&A]
I only want scalafmt to do a few basic things such as indentation-size correction and spacing between tokens like `=`. I am not a fan of the "Prettier" philosophy of dumping dozens of opinionated style corrections onto users that are difficult to exactly customize, justified by an insinuation that caring about optimizing style is "bike-shedding" and that aesthetic considerations don't not really matter. I think they do, and I want control over them in my codebase. But I would still like a formatter for obvious, un-opinionated corrections, like consistent indentation-size and EOF newlines.

Does scalafmt provide any way to disable all rules by default, so I can then specify which ones I want to be in effect? Or is there an alternative reliable auto-formatter I can use that will give me the control I want, closer to eslint than prettier?

**Comment by tgodzik** [comment]
You could experiment with the configuration to get to there, but I don't think there is currently a preset for that?

I see there was actually a thread on reddit about this https://www.reddit.com/r/scala/comments/19a4peq/is_there_a_minimalist_formatter/
